src/Core/Services/StateService.cs

```diff
@@ -67,6 +67,28 @@ public async Task<bool> IsAuthenticatedAsync(string userId = null)
             return await GetAccessTokenAsync(userId) != null;
         }
 
+        public async Task<string> GetUserIdAsync(string email)
+        {
+            if (email == null)
+            {
+                throw new Exception("email cannot be null");
+            }
+
+            await CheckStateAsync();
+            if (_state?.Accounts != null)
+            {
+                foreach (var account in _state.Accounts)
+                {
+                    var accountEmail = account.Value?.Profile?.Email;
+                    if (accountEmail != null && accountEmail == email)
```

Comment:
Given that `email` can't be `null` because of aforementioned check, then this can be directly written as 
`if (accountEmail == email)`